### PR TITLE
Basic signer and checker

### PR DESCRIPTION
**Pull request by malukenho**
- [x] Unit tests
- [x] ~AST checker?~

### CODE REVIEW

**Comment by malukenho** [comment]
@asgrim @Ocramius seems good to go?

File: src/CheckerInterface.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+interface CheckerInterface
+{
+    public function __construct(EncoderInterface $encoder);
+
+    /**
+     * @param string $phpCode
+     *
+     * @return bool
+     */
+    public function check(string $phpCode);
```
Comment (asgrim) [inline src/CheckerInterface.php]: As we're okay with PHP 7 here, use ` : bool` return declaration

File: src/CheckerInterface.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+declare(strict_types=1);
+
+namespace Roave\Signature;
+
+use Roave\Signature\Encoder\EncoderInterface;
+
+interface CheckerInterface
+{
+    public function __construct(EncoderInterface $encoder);
```
Comment (asgrim) [inline src/CheckerInterface.php]: I don't feel comfortable adding `__construct` to the interface here, I don't think this should be enforced...

File: src/FileContentChecker.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+     */
+    public function __construct(EncoderInterface $encoder)
+    {
+        $this->encoder = $encoder;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function check(string $phpCode)
```
Comment (asgrim) [inline src/FileContentChecker.php]: ` : bool`

File: src/FileContentChecker.php [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+        return $this->encoder->verify($this->stripCodeSignature($phpCode), $matches[1]);
+    }
+
+    /**
+     * @param string $phpCode
+     *
+     * @return string
+     */
+    private function stripCodeSignature(string $phpCode)
```
Comment (asgrim) [inline src/FileContentChecker.php]: ` : string`

File: src/SignerInterface.php [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+declare(strict_types=1);
+
+namespace Roave\Signature;
+
+use Roave\Signature\Encoder\EncoderInterface;
+
+interface SignerInterface
+{
+    public function __construct(EncoderInterface $encoder);
```
Comment (asgrim) [inline src/SignerInterface.php]: Same as above regarding putting `__construct` in interface, I don't think it belongs...

**Comment by malukenho** [comment]
Closing due github issues

**Pull request closed without merging**